pkg/configmanager/configmanager.go

```diff
@@ -213,17 +213,24 @@ func addOrUpdateRouterConfig(routerConfig *v2.RouterConfiguration) bool {
 	if routerConfig == nil {
 		return false
 	}
-	// support only one server
-	routers := config.Servers[0].Routers
-	for idx, rt := range routers {
-		if rt.RouterConfigName == routerConfig.RouterConfigName {
-			config.Servers[0].Routers[idx] = routerConfig
-			return true
+
+	// easy for test (config.Servers maybe equals nil),
+	// for example, a registry push does not trigger the dump store
+	if config.Servers != nil {
```

Comment:
if config.Servers != nil {
     return false
}

... other logic

Reply:
ok , fixed it.